kvdb/postgres/fixture.go

```diff
@@ -4,87 +4,123 @@ package postgres
 
 import (
 	"context"
+	"crypto/rand"
 	"database/sql"
-	"testing"
+	"encoding/hex"
+	"fmt"
+	"strings"
 
 	"github.com/btcsuite/btcwallet/walletdb"
 	embeddedpostgres "github.com/fergusstrange/embedded-postgres"
-	"github.com/stretchr/testify/require"
 )
 
 const (
-	testDsn = "postgres://postgres:postgres@localhost:9876/postgres?sslmode=disable"
-	prefix  = "test"
+	testDsnTemplate = "postgres://postgres:postgres@localhost:9876/%v?sslmode=disable"
+	prefix          = "test"
 )
 
-func clearTestDb(t *testing.T) {
-	dbConn, err := sql.Open("pgx", testDsn)
-	require.NoError(t, err)
+func getTestDsn(dbName string) string {
+	return fmt.Sprintf(testDsnTemplate, dbName)
+}
+
+var testPostgres *embeddedpostgres.EmbeddedPostgres
```

Comment:
Would be nice to wrap this in a struct with a `Start()` and `Stop()` method to not have global variables/functions.

Reply:
Will leave that for another time. I did change the start function to return a close closure, so the code has moved a little into that direction.